lib/node_modules/@stdlib/blas/base/dtrsm/lib/base.js

```diff
@@ -0,0 +1,425 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MAIN //
+
+/**
+* Solves matrix equation `op(A) * X = alpha * B` or `X * op(A) = alpha * B` where `alpha` is a scalar, `X` and `B` are m by n matrices, `A` is a unit, or non-unit,  upper or lower triangular matrix  and  `op( A )`  is one  of `op( A) = A` or `op( A ) = A**T`. The matrix `X` is overwritten on `B`.
+*
+* @private
+* @param {string} order - specifies the memory layout of `A` and `B`
+* @param {string} side - specifies whether `op( A )` appears on the left or right of `X`
+* @param {string} uplo - specifies whether the matrix `A` is an upper or lower triangular matrix
+* @param {string} transa - specifies the form of `op( A )` to be used in matrix multiplication
+* @param {string} diag - specifies whether or not `A` is unit triangular
+* @param {NonNegativeInteger} m - number of rows in `B`
+* @param {NonNegativeInteger} n - number of columns in `B`
+* @param {number} alpha - scalar alpha
+* @param {Float64Array} A - input matrix `A`
+* @param {NonNegativeInteger} LDA - stride of the first dimension of `A` (a.k.a., leading dimension of the matrix `A`)
+* @param {NonNegativeInteger} offsetA - index offset for matrix `A`
+* @param {Float64Array} B - input matrix `B`
+* @param {NonNegativeInteger} LDB - stride of the first dimension of `B` (a.k.a., leading dimension of the matrix `B`)
+* @param {NonNegativeInteger} offsetB - index offset for matrix `B`
+* @returns {Float64Array} matrix `B`
+*
+* @example
+* TODO:
+*/
+function dtrsm( order, side, uplo, transa, diag, m, n, alpha, A, LDA, offsetA, B, LDB, offsetB ) { // eslint-disable-line max-len
+	var nounit;
+	var lside;
+	var upper;
+	var tmp;
+	var i;
+	var j;
+	var k;
+
+	lside = side === 'L';
+	nounit = diag === 'N';
+	upper = uplo === 'U';
+
+	if ( m === 0 || n === 0 ) {
+		return B;
+	}
+	if ( alpha === 0.0 ) {
+		if ( order === 'row-major' ) {
+			for ( j = 0; j < n; j++ ) {
+				for ( i = 0; i < m; i++ ) {
+					B[ offsetB + ( i * LDB ) + j ] = 0.0;
+				}
+			}
+			return B;
+		}
+		for ( j = 0; j < n; j++ ) {
+			for ( i = 0; i < m; i++ ) {
+				B[ offsetB + ( j * LDB ) + i ] = 0.0;
+			}
+		}
+		return B;
+	}
+	if ( lside ) {
```

Comment:
It's likely that from here down, you can consolidate some of the loops and avoid explicit order branches by recognizing certain equivalences. E.g., when upper triangular column-major, equivalent of lower triangular row-major, etc.